bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/launch/LaunchUtils.java

```diff
@@ -122,10 +125,31 @@ private static void loadConfigurations(Map<String, String> configArgs, String co
                 ((BLogManager) logManager).loadUserProvidedLogConfiguration();
             }
 
-            boolean observeFlag = ConfigRegistry.getInstance().getAsBoolean(CONFIG_OBSERVABILITY_ENABLED);
-            if (observeFlag) {
-                ConfigRegistry.getInstance().addConfiguration(CONFIG_METRICS_ENABLED, Boolean.TRUE);
-                ConfigRegistry.getInstance().addConfiguration(CONFIG_TRACING_ENABLED, Boolean.TRUE);
+            if (ConfigRegistry.getInstance().contains(CONFIG_OBSERVABILITY_ENABLED)) {
```

Comment:
Shall we just extract a variable for this `ConfigRegistry.getInstance()`

Reply:
Fixed in 0420a885f7cd6f85221a5204378c0c3f791cf5ec